### PR TITLE
fix(list): Only apply aria-orientation with role

### LINKED ISSUE #944: List applies invalid default aria-orientation attribute
[mdc-list documentation](https://github.com/material-components/material-components-web/tree/master/packages/mdc-list#accessibility) says

> Use role="listbox" only for single selection list, without this role the ul element is implicitely role="list". Do not use aria-orientation attribute for standard list (i.e., role="list"), use component's vertical property to set the orientation to vertical.

The React wrapper applies `aria-orientation="vertical"` in default props: https://github.com/material-components/material-components-web-react/blob/master/packages/list/index.tsx#L117

I think the ideal fix is to refactor the `aria-orientation` prop to `orientation` (enum) or `vertical` (bool) and apply this as `aria-orientation` for allowed role values. But, this would be a breaking change.

This could be fixed without renaming props but it would be a little strange to pass an aria prop that is not passed to the root element. Maybe edgy enough not to worry about.

### PR DESCRIPTION
**Pull request by liamcmitchell-sc**
Fixes #944

### CODE REVIEW

File: packages/list/index.tsx [hunk 1]
```diff
@@ -146,9 +146,9 @@ export default class List extends React.Component<ListProps, ListState> {
     if (wrapFocus !== prevProps.wrapFocus) {
       this.foundation.setWrapFocus(wrapFocus!);
     }
-    if (this.props[ARIA_ORIENTATION] !== prevProps[ARIA_ORIENTATION]) {
+    if (this.props.orientation !== prevProps.orientation) {
       this.foundation.setVerticalOrientation(
-        this.props[ARIA_ORIENTATION] === VERTICAL
+        this.props.orientation !== HORIZONTAL
```
Comment (moog16) [inline packages/list/index.tsx]: Why not use the affirmative here? Meaning `=== VERTICAL`?

File: packages/list/index.tsx [hunk 2]
```diff
@@ -129,7 +129,7 @@ export default class List extends React.Component<ListProps, ListState> {
     }
     this.foundation.setWrapFocus(wrapFocus!);
     this.foundation.setVerticalOrientation(
-      this.props[ARIA_ORIENTATION] === VERTICAL
+      this.props.orientation !== HORIZONTAL
```
Comment (moog16) [inline packages/list/index.tsx]: Please leave a comment here as to why you're not using the affirmative (`=== VERTICAL`)

File: test/unit/list/index.test.tsx [hunk 3]
```diff
@@ -434,6 +432,43 @@ test('renders with role=menu if props.role=menu and props.checkboxList', () => {
...
+  const wrapper = shallow<List>(<List>{children}</List>);
+  assert.equal(wrapper.props()['aria-orientation'], undefined);
+});
+
+test('renders without aria-orientation when checkboxList and orientation=vertical (default)', () => {
+  const wrapper = shallow<List>(<List checkboxList>{children}</List>);
+  assert.equal(wrapper.props()['aria-orientation'], undefined);
+});
+
+test('renders with aria-orientation when checkboxList and orientation=horizontal', () => {
```
Comment (moog16) [inline test/unit/list/index.test.tsx]: Can you add an equivalent test, but swap `checkboxList` --> `radioList`

**Comment by liamcmitchell-sc** [comment]
Done. Yes, tested this in my project. Works as expected.

**Comment by moog16** [comment]
Thanks!

**Pull request merged**